tendrl/node_agent/persistence/disk.py

```diff
@@ -0,0 +1,31 @@
+from tendrl.common.etcdobj.etcdobj import EtcdObj
+from tendrl.common.etcdobj import fields
+
+
+class Disk(EtcdObj):
+    """A table of the Disk, lazily updated
+
+    """
+    __name__ = 'nodes/%s/Disk/%s'
+
+    node_id = fields.StrField("node_id")
+    device_name = fields.StrField("device_name")
+    fs_type = fields.StrField("fs_type")
+    fs_uuid = fields.StrField("fs_uuid")
+    model = fields.StrField("model")
+    mount_point = fields.StrField("mount_point")
+    name = fields.StrField("name")
+    parent = fields.StrField("parent")
+    size = fields.StrField("size")
+    type = fields.StrField("type")
+    used = fields.StrField("used")
+    ssd = fields.StrField("ssd")
+    vendor = fields.StrField("vendor")
+    disk_id = fields.StrField("disk_id")
+
+    def render(self):
+        self.__name__ = self.__name__ % (
+            self.node_id,
+            (self.device_name.replace("/", "", 1)).replace("/", "_")
```

Comment:
Disk id will come only when it is used, if the disk is not used then it does not have any id, so we can use disk name only